core/kroma_migration.go

```diff
@@ -0,0 +1,197 @@
+package core
+
+import (
+	"bytes"
+	"encoding/binary"
+	"encoding/gob"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethdb"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+var HaltOnStateTransition = errors.New("historical rpc must be set to transition to MPT")
+
+type MigratedRef struct {
+	db ethdb.Database
+	mu sync.Mutex
+
+	root   common.Hash
+	number uint64
+}
+
+var (
+	accountChangesPrefix  = []byte("aC-")
+	storageChangesPrefix  = []byte("sC-")
+	destructChangesPrefix = []byte("dC-")
+	migratedRootKey       = []byte("MigratedRoot")
+	migratedNumberKey     = []byte("MigratedNumber")
+)
```

Comment:
I think it's better to move these declarations above of MigratedRef declaration.